### PR TITLE
boards/pba-d-01-kw2x: add network interface auto init

### PR DESCRIPTION
**Pull request by jfischer-phytec-iot**
This PR adds network interface auto init as it was introduced in ~~#2891~~ #2901.

### CODE REVIEW

**Comment by kaspar030** [comment]
Damn, #2901 seems stalled in reviewing...

**Comment by jremmert-phytec-iot** [comment]
I wasn´t sure which one of the auto-init methods will be the standard one. The existing merged #2891 or #2901. Therefore I choosed #2891, since this is already merged. I have no strong opinion there.

**Comment by jfischer-phytec-iot** [comment]
updated for #2901

File: boards/pba-d-01-kw2x/Makefile.dep [hunk 1]
```diff
@@ -0,0 +1,4 @@
+ifneq (,$(filter ng_netif,$(USEMODULE)))
+  USEMODULE += kw2xrf
```
Comment (kaspar030) [inline boards/pba-d-01-kw2x/Makefile.dep]: don't we do the dependencies the other way around? see #2947.

Comment (jfischer-phytec-iot) [inline boards/pba-d-01-kw2x/Makefile.dep]: no

Comment (jremmert-phytec-iot) [inline boards/pba-d-01-kw2x/Makefile.dep]: sry, I don´t got the point correctly. What would be the alternative? You mean ```  USEMODULE += ng_ieee802154``` is missing here?

Comment (gebart) [inline boards/pba-d-01-kw2x/Makefile.dep]:
```
ifneq (,$(filter kw2xrf,$(USEMODULE)))
  USEMODULE+=ng_netif
```

Comment (jremmert-phytec-iot) [inline boards/pba-d-01-kw2x/Makefile.dep]: so, we should add @gebart s snip to the RIOT/Makefile.dep? That would make this PR dependant of #2947. But the ```boards/pba-d-01-kw2x/Makefile.dep``` can remain as is?

Comment (jfischer-phytec-iot) [inline boards/pba-d-01-kw2x/Makefile.dep]: >don't we do the dependencies the other way around? see #2947.

It's not even merged.

>ifneq (,$(filter kw2xrf,$(USEMODULE)))
  USEMODULE+=ng_ieee802154

Where to put it?

Please write in clear sentences, not in riddles.

Comment (kaspar030) [inline boards/pba-d-01-kw2x/Makefile.dep]: sorry for the confusion!
<pre>
ifneq (,$(filter kw2xrf,$(USEMODULE)))
USEMODULE+=ng_ieee802154
</pre>
should go in mkefile.ep, but we can fix that once #2947 is merged. the concept is not fixed, yet.

**Comment by jremmert-phytec-iot** [comment]
ok, then I would vote for merging this first. #2947 needs travis check and this one is ready so far. Maybe @authmillenon could add this to #2947 before rebasing?

Do you ACK?

**Comment by OlegHahm** [comment]
Works like charm. ACK

**Pull request merged**